src/usr/local/www/interfaces_vxlan_edit.php

```diff
@@ -0,0 +1,250 @@
+<?php
+/*
+ * interfaces_vxlan_edit.php
+ *
+ * part of pfSense (https://www.pfsense.org)
+ * Copyright (c) 2004-2013 BSD Perimeter
+ * Copyright (c) 2013-2016 Electric Sheep Fencing
+ * Copyright (c) 2014-2020 Rubicon Communications, LLC (Netgate)
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+##|+PRIV
+##|*IDENT=page-interfaces-vxlan-edit
+##|*NAME=Interfaces: VXLAN: Edit
+##|*DESCR=Allow access to the 'Interfaces: VXLAN: Edit' page.
+##|*MATCH=interfaces_vxlan_edit.php*
+##|-PRIV
+
+require_once("guiconfig.inc");
+
+init_config_arr(array('vxlans', 'vxlan'));
+$a_vxlans = &$config['vxlans']['vxlan'];
+$id = $_REQUEST['id'];
+
+
+if (isset($id) && $a_vxlans[$id]) {
+	$pconfig['if'] = $a_vxlans[$id]['if'];
+	if (!empty($a_vxlans[$id]['ipaddr'])) {
+		$pconfig['if'] = $pconfig['if'] . '|' . $a_vxlans[$id]['ipaddr'];
+	}
+	$pconfig['vxlanif'] = $a_vxlans[$id]['vxlanif'];
+	$pconfig['remote-addr'] = $a_vxlans[$id]['remote-addr'];
+	$pconfig['vxlanid'] = $a_vxlans[$id]['vxlanid'];
+	$pconfig['vxlanlocalport'] = $a_vxlans[$id]['vxlanlocalport'];
+	$pconfig['vxlanremoteport'] = $a_vxlans[$id]['vxlanremoteport'];
+	$pconfig['vxlanttl'] = $a_vxlans[$id]['vxlanttl'];
+	$pconfig['vxlanlearn'] = isset($a_vxlans[$id]['vxlanlearn']);
+	$pconfig['descr'] = $a_vxlans[$id]['descr'];
+} else {
+	$pconfig['vxlanlearn'] = true;
+}
+
+if ($_POST['save']) {
+	unset($input_errors);
+	$pconfig = $_POST;
+
+	/* input validation */
+	$reqdfields = explode(" ", "if remote-addr vxlanid");
+	$reqdfieldsn = array(gettext("Parent interface"), gettext("VXLAN remote address"), gettext("VXLAN ID"));
+
+	do_input_validation($_POST, $reqdfields, $reqdfieldsn, $input_errors);
+
+	if (!is_ipaddr($_POST['remote-addr']) || is_subnet($_POST['remote-addr'])) {
+		$input_errors[] = gettext("The remote peer address must have valid IP address and must not contain CIDR masks or prefixes.");
+	}
+
+	$alias = strstr($_POST['if'], '|');
+	if ((is_ipaddrv4($alias) && !is_ipaddrv4($_POST['remote-addr'])) ||
+	    (is_ipaddrv6($alias) && !is_ipaddrv6($_POST['remote-addr']))) {
+		$input_errors[] = gettext("The alias IP address family has to match the family of the remote peer address.");
+	}
+
+	if (!is_numeric($_POST['vxlanid']) || ($_POST['vxlanid'] < 0) || ($_POST['vxlanid'] > 16777216)) {
+		$input_errors[] = gettext("VXLAN ID must be in range between 0 and 16777216.");
```

Comment:
Since `0` and `16777216` are not valid values, this message is ambiguous (it is missing a qualifier such as "exclusive" for example).

It is more clear to the user to say the valid range is `1-16777215`.